### PR TITLE
DGX inherit from LocalCUDACluster

### PR DESCRIPTION
**Pull request by madsbk**
This PR makes DGX inherit from `LocalCUDACluster`, which sets `UCX_NET_DEVICES` for each worker.

### CODE REVIEW

File: dask_cuda/tests/test_dgx.py [hunk 1]
```diff
@@ -1,85 +1,122 @@
...
+            res = da.from_array(numpy.arange(10000), chunks=(1000,))
+            res = res.sum().compute()
+            assert res == 49995000
+
+
+def test_tcp_only():
+    p = mp.Process(target=_test_tcp_only)
+    p.start()
+    p.join()
+    assert not p.exitcode
```
Comment (pentschev) [inline dask_cuda/tests/test_dgx.py]: Do we really need this? It seems to be the same as `test_default`, given that by default TCP will be used.

Comment (madsbk) [reply dask_cuda/tests/test_dgx.py]: What would yo8u suggest is the default protocol? Right now it is `DGX(protocol="ucx", ...)`. Should it be `tcp` instread?

Comment (pentschev) [reply dask_cuda/tests/test_dgx.py]: Oh sorry, I have missed that the default has been changed.

I'm not sure what the default should be, it makes sense to have UCX, but I'm not sure if it's not too early to make it default. I think I would keep TCP as default for the time being, @mrocklin do you have a suggestion to untie?

Regardless of what's the default, I see that we need both tests, so let's keep them as they are now.

Comment (madsbk) [reply dask_cuda/tests/test_dgx.py]: I agree, I think it is best to keep TCP as the default for now. When we are more confident that UCX works out-of-the-box we can change it to the default.

**Comment by madsbk** [comment]
@pentschev @mrocklin, I think PR is ready for merging.

**Comment by pentschev** [comment]
Changes look good to me. Thanks for working on this @madsbk. Merging.

File: dask_cuda/tests/test_dgx.py (hunk 1 above)
Comment (mrocklin) [reply dask_cuda/tests/test_dgx.py]: > do you have a suggestion to untie?

Maybe this eventually?

```python
try:
    import ucp
except ImportError:
    protocol = "tcp"
else:
    protocol = "ucx"
```

Comment (madsbk) [reply dask_cuda/tests/test_dgx.py]: Yes

Comment (pentschev) [reply dask_cuda/tests/test_dgx.py]: Yeah, makes sense once we have stable packages and can make dask-cuda require them.

Comment (pentschev) [reply dask_cuda/tests/test_dgx.py]: Urm, nevermind. The suggestions is to make protocol _if_ ucp is available. That makes more sense.

**Pull request merged**